csrc/host_ir/lower_to_communication.cpp

```diff
@@ -523,8 +523,30 @@ std::vector<Expr*> convertSingleOpToCommunication(
       "Resharding on an inner axis is not lowerable ",
       e->toString());
 
-  if (auto* reduce = dynamic_cast<ReductionOp*>(e)) {
-    BinaryOpType op_type = reduce->getReductionOpType();
+  if (auto* load_store = dynamic_cast<LoadStoreOp*>(e)) {
```

Comment:
```suggestion
  if (dynamic_cast<LoadStoreOp*>(e) != nullptr) {
```
**[Suggested by AI]** The logic is still “execute the body only when e can be cast to LoadStoreOp”.  
What changed is the way this test is written:

• Before: it relied on the C++17 “if-with-initializer” syntax and created a
  local pointer variable (load_store) that wasn’t used later.

• After: it simply performs the cast and compares the result with nullptr,
  avoiding the extra variable and using syntax that is valid in older C++
  standards.

Thus the patch only removes the build-breaking construct/unused variable;
behaviour of the program is unchanged.